react/src/hooks/useFeatureFlags.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useCallback } from 'react'
+import { usePostHogContext, FeatureFlags } from '../context'
+
+/**
+ * A hook that fetches active feature flags and determines which flags are enabled for the user.
+ * @param props.refreshInterval - How often to refresh the feature flags, in seconds.
+ * @param props.sendEvent - A flag that controls whether an event will be sent on flag refresh.
+ * @returns An object containing flags that are enabled for the user.
+ */
+export function useFeatureFlags(props: { refreshInterval: number; sendEvent: boolean }): FeatureFlags['enabled'] {
+    const { refreshInterval = 0, sendEvent = true } = props || {}
+    const { client: posthog, featureFlags, setFeatureFlags } = usePostHogContext()
+
+    const getFeatureFlags = useCallback((): void => {
+        if (posthog) {
+            const active = featureFlags.active || posthog.featureFlags.getFlags()
+
+            const enabled = active.reduce((result: FeatureFlags['enabled'], flag: string) => {
+                result[flag] = !!posthog.isFeatureEnabled(flag, {
+                    send_event: sendEvent,
+                })
+                return result
+            }, {})
+
+            setFeatureFlags({ active, enabled })
+        }
+    }, [posthog, featureFlags.active, sendEvent, setFeatureFlags])
+
+    useEffect(() => {
+        if (posthog && !featureFlags.active) getFeatureFlags()
+    }, [posthog, featureFlags.active, getFeatureFlags])
+
+    useEffect(() => {
```

Comment:
I think this can be dropped and we can instead use `posthog.onFeatureFlags`.

Reply:
Thanks for your comments! I've modified the hook to use the `onFeatureFlags` event handler, and also modified the interval to use the `reloadFeatureFlags()` method so that changes to currently active flags are correctly propagated.

Regarding where to make a home for this posthog-js-react package, I personally think it makes more sense to place it under the control of the PostHog GitHub account. I think there is a great opportunity to expand out your React tooling, and I would be more than happy to continue contributing. As well, I would imagine that you want control of npm publishing for this too. Let me know if you would prefer the alternative and I will set things up.